wagtail/core/models.py

```diff
@@ -2758,14 +2758,17 @@ def update(self, user=None, next_task=None):
                     # not STATUS_IN_PROGRESS), move to the next task
                     self.current_task_state = next_task.specific.start(self, user=user)
                     self.save()
+                    # if task has auto-approved, update the workflow again
+                    if self.current_task_state.status != self.current_task_state.STATUS_IN_PROGRESS:
+                        self.update(user=user)
                 # otherwise, continue on the current task
             else:
                 # if there is no uncompleted task, finish the workflow.
                 self.finish(user=user)
 
     def get_next_task(self):
         """Returns the next active task associated with the latest page revision, which has not been either approved or skipped"""
-        return Task.objects.filter(workflow_tasks__workflow=self.workflow, active=True).exclude(Q(task_states__page_revision=self.page.get_latest_revision()), Q(task_states__status=TaskState.STATUS_APPROVED) | Q(task_states__status=TaskState.STATUS_SKIPPED)).order_by('workflow_tasks__sort_order').first()
+        return Task.objects.filter(workflow_tasks__workflow=self.workflow, active=True).exclude(task_states__in=TaskState.objects.filter(Q(page_revision=self.page.get_latest_revision()), Q(status=TaskState.STATUS_APPROVED) | Q(status=TaskState.STATUS_SKIPPED))).order_by('workflow_tasks__sort_order').first()
```

Comment:
This could really do with being split onto multiple lines...
```suggestion
        return (
            Task.objects.filter(workflow_tasks__workflow=self.workflow, active=True)
            .exclude(
                task_states__in=TaskState.objects.filter(
                    Q(page_revision=self.page.get_latest_revision()),
                    Q(status=TaskState.STATUS_APPROVED) | Q(status=TaskState.STATUS_SKIPPED)
                )
            ).order_by('workflow_tasks__sort_order').first()
        )
```

Reply:
Good point, thanks! Happy to commit that